test/test_autograd.py

```diff
@@ -1020,44 +1016,45 @@ def reset_grad():
         reset_grad()
         torch.autograd.backward(fn(), gradient, inputs=[x])
         self.assertEqual(x.grad, x_grad_expected)
-        self.assertEqual(y.grad, torch.zeros(2, 2))
+        self.assertEqual(y.grad, torch.zeros(2, 2, dtype=torch.double))
```

Comment:
nit: do we prefer here to pass `exact_dtype=False` here instead of creating double Tensors?
Same for a few below.